File: examples/async/bitget_ws_private_readonly.py
"""Open a Bitget private WebSocket user-data stream."""

import asyncio
import os

from dcex.ws import bitget


async def main() -> None:
    """Open the private stream, subscribe to account updates, and print one message."""
    api_key = os.environ["BITGET_API_KEY"]
    api_secret = os.environ["BITGET_API_SECRET"]
    passphrase = os.environ["BITGET_PASSPHRASE"]

    async with bitget.private(
        api_key=api_key,
        api_secret=api_secret,
        passphrase=passphrase,
    ) as ws:
        print(await ws.recv())
        await ws.subscribe_account()
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/bitget_ws_private_readonly.rs
use std::env;
use std::time::Duration;

use dcex::ws::bitget::BitgetPrivateWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let api_key = env::var("BITGET_API_KEY")
        .map_err(|error| dcex::DcexError::InvalidInput(error.to_string()))?;
    let api_secret = env::var("BITGET_API_SECRET")
        .map_err(|error| dcex::DcexError::InvalidInput(error.to_string()))?;
    let passphrase = env::var("BITGET_PASSPHRASE")
        .map_err(|error| dcex::DcexError::InvalidInput(error.to_string()))?;

    let mut ws =
        BitgetPrivateWebSocket::new(api_key, api_secret, passphrase, Duration::from_secs(10))?;
    ws.connect().await?;
    println!("{}", ws.recv().await?);
    ws.subscribe_account("USDT-FUTURES").await?;
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
